### PR TITLE
fix(trustlab): enforce role-based Payload content access

### PR DESCRIPTION
**Pull request by kilemensi**
## Description

`@/trustlab`'s Payload access layer had two categories of issues.

**Authorization gaps** (P1):
- Reports, Playbooks, and Toolkits used `Boolean(user)`, so any authenticated session, regardless of role, could create, update, or delete these collections.
- Partners, Donors, Opportunities, and Organisations had no explicit `create`/`update`/`delete` access at all, relying on Payload's implicit behaviour.
- `SiteSettings` was world-readable: `loggedIn` was defined as `(user) => Boolean(user)` but wired as a Payload access adapter, so Payload called it as `loggedIn({ req })`, which was always truthy.

**Structural issues** that made auditing difficult:
- User-level predicates and Payload access adapters shared the same function signatures, so call sites had to inline `({ req: { user } }) => helper(user)` everywhere.
- `Boolean(user)` trusts any truthy `req.user` value; a token with an arbitrary payload would pass.

### Changes

**Access layer refactor (`src/payload/access/`)**

Introduces a clear three-tier naming convention:

| Tier | Returns | Used by |
|---|---|---|
| `isXxx(user)` | `boolean` | custom route/middleware code |
| `canXxx(user)` | `boolean \| Payload where-filter` | internal policy logic |
| `hasXxxAccess({ req })` | Payload access adapter | collection/global `access:` config |

- `hasValidRole(user)` added to `roles.js`: validates `user.role` against the known role constants. Replaces `Boolean(user)` everywhere.
- `loggedIn.js` removed. `isLoggedIn`, `isEditor`, `isAuthor`, `isAdmin`, `canAuthor`, `canManageUser` are the new user-level predicates.
- `hasLoggedInAccess`, `hasEditorAccess`, `hasAuthorAccess`, `hasAdminAccess`, `hasManageUserAccess`, `hasCreateUserAccess` are the new Payload adapters, each a thin wrapper over the corresponding predicate.

**Collections and globals**

| Surface | Before | After |
|---|---|---|
| Reports, Playbooks, Toolkits (write) | `Boolean(user)` | `hasEditorAccess` |
| Pages (write) | `canManagePages(user)` inline | `hasEditorAccess` |
| Opportunities, Organisations, Partners, Donors (write) | missing (implicit) | `hasEditorAccess` |
| Posts, Media (write) | `canManageContent(user)` inline | `hasAuthorAccess` |
| Users collection | inline wrappers | named adapters; role-field update tightened to `hasCreateUserAccess` |
| SiteSettings read | `loggedIn` (broken — always `true`) | `hasLoggedInAccess` |
| SiteSettings update | `canManageSiteSettings` inline | `hasAdminAccess` |

Posts and Media intentionally remain on `hasAuthorAccess` to preserve the ownership-aware author workflow.

**Preview route**

`preview/route.ts` now uses `isAuthor(user)`, the correct boolean predicate, instead of the old `canManageContent(user)` whose filter-object return value was only coincidentally truthy in a boolean context.

**Tests (`src/payload/access/index.test.js`)**

Full policy matrix at the helper layer: `isAuthor`, `canAuthor`, `hasAuthorAccess`, `hasEditorAccess`, `hasAdminAccess`, `hasLoggedInAccess`, `hasCreateUserAccess`, `hasManageUserAccess`, and `anyone`. Cases cover admin/editor/author/anonymous and invalid-role shapes (`role: "unknown"`, bare integer, `undefined`). The `canAuthor` suite includes the id-guard edge case (author with no `id` returns `false` rather than a filter with `equals: undefined`).

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Screenshots

N/A

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by kilemensi** [comment]
@codex review

File: apps/trustlab/src/payload/globals/index.js [hunk 1]
```diff
@@ -14,8 +16,8 @@ const SiteSettings = {
   access: {
     // Since we're using Local APIs, we should still be able to pull data server-side
     // See: note in https://payloadcms.com/docs/local-api/overview#transactions
-    read: loggedIn,
-    update: canManageSiteSettings,
+    read: hasLoggedInAccess,
```
[the remark this answers is not in the document]
Comment (kilemensi) [reply apps/trustlab/src/payload/globals/index.js]: Fixed. @codex review

**Comment by kilemensi** [comment]
@claude review

**Comment by kilemensi** [comment]
[P2] canManageUser skips the hasValidRole guard and [P3] Missing edge-case tests for hasManageUserAccess without id are now fixed.

On isAuthor naming vs semantics, this matches isEditor which also returns true for editors and admins.

@claude review

File: apps/trustlab/src/payload/access/abilities.js [hunk 2]
```diff
@@ -1,39 +1,57 @@
...
+
+// Admins and editors can manage any content
+export const isEditor = (user) => checkRole([ROLE_ADMIN, ROLE_EDITOR], user);
+
+export const hasEditorAccess = ({ req } = {}) => isEditor(req?.user);
+
+export const isAuthor = (user) => {
+  // Any valid CMS user has at least author-level capability.
+  return isLoggedIn(user);
+};
```
Comment (koechkevin) [inline apps/trustlab/src/payload/access/abilities.js:16]: I think this `isAuthor` name is misleading/redundant if its functionality is identical to `isLoggedIn`

**Comment by kilemensi** [comment]
Going twice @CodeForAfrica/tech ...

**Comment by koechkevin** [comment]
> Going twice @CodeForAfrica/tech ...

Currently testing @kilemensi

**Review by koechkevin — APPROVED** [review]
Looks good to me

File: apps/trustlab/src/payload/access/abilities.js (hunk 2 above)
Comment (kilemensi) [reply apps/trustlab/src/payload/access/abilities.js:16]: My understanding of the code is they're different @koechkevin :

  * `isLoggedIn`: is used to check if the current user is anonymous or authenticated i.e. handles the first A in [AAA](https://en.wikipedia.org/wiki/Authentication,_authorization,_and_accounting).
  * `isAuthor` (and `isEditor`, `isAdmin`, etc.): is used to check authorisation level that should be given to the authenticated user i.e. the second A in AAA. The fact this two functions mean the same right now is just a coincidence; if we introduce say `contributor` role to the app, this would no longer be the case.
